context/parseLambda.ts

```diff
@@ -171,110 +228,156 @@ export const parseLambda = (
         data: delegateData ?? {},
       },
     ];
-
-  const contractAddress = (() => {
-    const expr = lambda.find(expr => {
-      if (!("prim" in expr)) return false;
-
-      const { prim, args } = expr;
-
-      //@ts-expect-error
-      if (prim !== "PUSH" || args?.[0]?.prim !== "address") return false;
-
-      //@ts-expect-error
-      if (!!args?.[1]?.bytes) {
-        return (
-          validateAddress(
-            encodePubKey(
+  } else {
+    const [isUnDelegate, type] = parseUnDelegate(lambda);
+    if (isUnDelegate) {
+      return [
+        type,
+        {
+          contractAddress: "",
+          mutez: undefined,
+          entrypoint: {
+            name: "",
+            params: { name: "", type: "" },
+          },
+          data: {},
+        },
+      ];
+    }
+  }
+
+  // FA1.2 and FA2 is a special case of contract_execution. There number of instrucions is the same as regular contract execution
+  const contract_execution_instr_size = 7;
+  if (lambda.length != contract_execution_instr_size)
+    return [LambdaType.LAMBDA_EXECUTION, undefined];
+
+  const fail_parse: [boolean, undefined] = [false, undefined];
+  const succ_parse: [boolean, undefined] = [true, undefined];
+
+  // parse DROP
+  const [parseDrop] = parsePrimPattern(lambda, 0, "DROP", () => succ_parse);
```

Comment:
There's all this variable names: `parseDrop`, `parsePushAddr`, `parseMutez`, etc... Imo the naming is a bit weird because it looks like everything is a function name, which makes it difficult to understand the code. Maybe something like: `couldParseDrop` or something similar can help to make it more clear

Reply:
modified.